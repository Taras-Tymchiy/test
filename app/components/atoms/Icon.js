// @flow
import * as React from 'react';
import {
    Image
  } from 'react-native';
import { COLOR_IOS_BUTTON_TEXT, FONT_SIZE_M, ICON_SIZE, ICON_OPACITY } from '../styles/constants';

export interface IconProps {
    +source: any;
    +size?: number;
}

export default function Icon({source, size}: IconProps) {
  size = size || ICON_SIZE;
  const opacity = ICON_OPACITY;
  return (
    <Image source={source} style={{width: size, height: size, opacity}} resizeMode='center' resizeMethod='scale' />
  );
};
