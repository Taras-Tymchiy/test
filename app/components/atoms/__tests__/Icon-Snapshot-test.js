import React from 'react';
import renderer from 'react-test-renderer';
import Icon, { IconProps } from '../Icon';

const props: IconProps = {
    sourse: 'test source'
};

it('renders an Icon using Snapshots', () => {
  expect(renderer.create(
    <Icon {...props} />
  )).toMatchSnapshot();
});

it('renders an Icon with custom size using Snapshots', () => {
  const newProps = {
    ...props,
    size: 33
  };
  expect(renderer.create(
    <Icon {...newProps} />
  )).toMatchSnapshot();
});
