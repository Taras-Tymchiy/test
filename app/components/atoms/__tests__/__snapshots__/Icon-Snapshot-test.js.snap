// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an Icon using Snapshots 1`] = `
<Image
  resizeMethod="scale"
  resizeMode="center"
  source={undefined}
  style={
    Object {
      "height": 16,
      "opacity": 0.5,
      "width": 16,
    }
  }
/>
`;

exports[`renders an Icon with custom size using Snapshots 1`] = `
<Image
  resizeMethod="scale"
  resizeMode="center"
  source={undefined}
  style={
    Object {
      "height": 33,
      "opacity": 0.5,
      "width": 33,
    }
  }
/>
`;
